src/torrent.cpp

```diff
@@ -9239,14 +9239,14 @@ namespace libtorrent
 	void torrent::set_queue_position(int p)
 	{
 		TORRENT_ASSERT(is_single_thread());
-		TORRENT_ASSERT((p == -1) == is_finished()
-			|| (!m_auto_managed && p == -1)
-			|| (m_abort && p == -1));
-		if (is_finished() && p != -1) return;
```

Comment:
this early exit condition seems to still make sense. was there a case where this would break anything?